secboot/secboot_sb_test.go

```diff
@@ -1286,23 +1176,18 @@ func (s *secbootSuite) TestUnlockVolumeUsingSealedKeyIfEncryptedFdeRevealKeyErr(
 
 	opts := &secboot.UnlockVolumeUsingSealedKeyOptions{}
 	_, err = secboot.UnlockVolumeUsingSealedKeyIfEncrypted(mockDiskWithEncDev, defaultDevice, mockSealedKeyFile, opts)
-	c.Assert(err, ErrorMatches, `(?s)cannot run fde-reveal-key: 
------
-failed
-service result: exit-code
------`)
-	// ensure no tmp files are left behind
-	c.Check(osutil.FileExists(filepath.Join(dirs.GlobalRootDir, "/run/fde-reveal-key")), Equals, false)
+	c.Assert(err, ErrorMatches, `cannot run fde-reveal-key "reveal": helper error`)
 }
 
 func (s *secbootSuite) TestUnlockVolumeUsingSealedKeyIfEncryptedFdeRevealKey(c *C) {
-	// this test uses a real systemd-run --user so check here if that
-	// actually works
-	if output, err := exec.Command("systemd-run", "--user", "--wait", "--collect", "--service-type=exec", "/bin/true").CombinedOutput(); err != nil {
-		c.Skip(fmt.Sprintf("systemd-run not working: %v", osutil.OutputErr(output, err)))
-	}
+	var reqs []*fde.RevealKeyRequest
+	restore := fde.MockRunFDERevealKey(func(req *fde.RevealKeyRequest) ([]byte, error) {
```

Comment:
Nice, this tests is so much simpler with this change and more to the point (i.e. do not require all the detail knowledge about fde details now)